### PR TITLE
vim-patch:8.1.1687: the evalfunc.c file is too big

### PR DESCRIPTION
**Pull request by lewis6991**
Problem:    The evalfunc.c file is too big.
Solution:   Move testing support to a separate file.
https://github.com/vim/vim/commit/ecaa70ea29c269dd0dabd3cd5acdfa0ce42ccd54

### CODE REVIEW

File: src/nvim/testing.c [hunk 1]
```diff
@@ -0,0 +1,555 @@
+#include "nvim/eval/encode.h"
```
Comment (zeertzjq) [inline src/nvim/testing.c:1]: Add a PVS header

Comment (lewis6991) [reply src/nvim/testing.c:1]: Thanks, keep forgetting that.

File: src/nvim/testing.c [hunk 2]
```diff
@@ -0,0 +1,558 @@
...
+  } else {
+    FILE *const fd2 = os_fopen(fname2, READBIN);
+    if (fd2 == NULL) {
+      fclose(fd1);
+      snprintf((char *)IObuff, IOSIZE, (char *)e_notread, fname2);
+    } else {
+      int64_t linecount = 1;
+      for (int64_t count = 0;; count++) {
+        const char c1 = (char)fgetc(fd1);
+        const char c2 = (char)fgetc(fd2);
```
Comment (zeertzjq) [inline src/nvim/testing.c:303]: I think making these two variables `char` may cause problems. Better to do the casting in the assignments below.

File: src/nvim/eval.c [hunk 3]
```diff
@@ -6056,328 +5916,6 @@ void assert_error(garray_T *gap)
...
-      assert_error(&ga);
-      ga_clear(&ga);
-      return 1;
-    }
-  }
-  return 0;
-}
-
-/// Common for assert_true() and assert_false().
-int assert_bool(typval_T *argvars, bool is_true)
```
Comment (zeertzjq) [inline src/nvim/eval.c:6221]: This function can be static.

File: src/nvim/testing.c [hunk 4]
```diff
@@ -0,0 +1,558 @@
...
+  rettv->vval.v_number = assert_match_common(argvars, ASSERT_MATCH);
+}
+
+/// "assert_notmatch(pattern, actual[, msg])" function
+void f_assert_notmatch(typval_T *argvars, typval_T *rettv, FunPtr fptr)
+{
+  rettv->vval.v_number = assert_match_common(argvars, ASSERT_NOTMATCH);
+}
+
+/// "assert_report(msg)
```
Comment (zeertzjq) [inline src/nvim/testing.c:523]: There is a typo here, but it is also present before this PR.

**Review by zeertzjq — APPROVED** [review]

File: src/nvim/testing.c [hunk 5]
```diff
@@ -0,0 +1,558 @@
...
+// it. PVS-Studio Static Code Analyzer for C, C++ and C#: http://www.viva64.com
+
+// testing.c: Support for tests
+
+#include "nvim/eval/encode.h"
+#include "nvim/ex_docmd.h"
+#include "nvim/eval.h"
+#include "nvim/os/os.h"
+#include "nvim/testing.h"
+
```
Comment (zeertzjq) [inline src/nvim/testing.c:11]: Include generated header

File: src/nvim/testing.c (hunk 4 above)
Comment (lewis6991) [reply src/nvim/testing.c:523]: what's the typo?

Comment (lewis6991) [reply src/nvim/testing.c:523]: ah, I see now.

**Review by zeertzjq — APPROVED** [review]

**Pull request merged**